src/canillita_newsflash.erl

```diff
@@ -0,0 +1,139 @@
+%%% @doc NewsFlash Model
+-module(canillita_newsflash).
+
+-behaviour(sumo_doc).
+-behaviour(sumo_rest_doc).
+
+-type id() :: binary().
+-type newspaper_name() :: binary().
+-type title() :: binary().
+-type body() :: binary().
+
+-opaque news_flash() ::
+  #{ id => id() | undefined
+   , newspaper_name => newspaper_name()
+   , title => title()
+   , body => body()
+   , created_at => calendar:datetime()
+   , updated_at => calendar:datetime()
+   }.
+
+-export_type(
+  [ id/0
+  , newspaper_name/0
+  , title/0
+  , body/0
+  , news_flash/0]
+  ).
+
+%% sumo_doc behaviour callbacks
+-export(
+  [ sumo_schema/0
+  , sumo_sleep/1
+  , sumo_wakeup/1
+  ]).
+
+%% sumo_rest_doc behaviour callbacks
+-export(
+  [ to_json/1
+  , from_json/1
+  , from_json/2
+  , update/2
+  , uri_path/1
+  ]).
+
+%% Public API
+-export(
+  [new/3]).
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% sumo_doc behaviour callbacks
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+-spec sumo_schema() -> sumo:schema().
+sumo_schema() ->
+  sumo:new_schema(
+    ?MODULE,
+    [ sumo:new_field(id, binary, [id, unique])
+    , sumo:new_field(newspaper_name, string, [not_null])
+    , sumo:new_field(title, string, [not_null])
+    , sumo:new_field(body, binary, [not_null])
+    , sumo:new_field(created_at, datetime, [not_null])
+    , sumo:new_field(updated_at, datetime, [not_null])
+    ]).
+
+-spec sumo_sleep(NewsFlash::news_flash()) -> sumo:doc().
+sumo_sleep(NewsFlash) -> NewsFlash.
+
+-spec sumo_wakeup(NewsFlash::sumo:doc()) -> news_flash().
+sumo_wakeup(NewsFlash) -> NewsFlash.
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% sumo_rest_doc behaviour callbacks
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+-spec to_json(NewsFlash::news_flash()) -> news_flash().
+to_json(NewsFlash) ->
+  #{ id => sr_json:encode_null(maps:get(id, NewsFlash))
+   , newspaper_name => maps:get(newspaper_name, NewsFlash)
+   , title => maps:get(title, NewsFlash)
+   , body => maps:get(body, NewsFlash)
+   , created_at => sr_json:encode_date(maps:get(created_at, NewsFlash))
+   , updated_at => sr_json:encode_date(maps:get(updated_at, NewsFlash))
+   }.
+
+-spec from_json(NewspaperName::newspaper_name(), Json::sumo_rest_doc:json()) ->
+  {ok, news_flash()} | {error, iodata()}.
+from_json(NewspaperName, Json) ->
```

Comment:
That's not how `from_json/2` is expected to work according to `sumo_rest_doc`, but I'm not sure if what you are doing here is actually wrong. Let's discuss it with the team tomorrow